### DISCUSSION TITLE
No flows showing - 'flow-LAABIGYMRYZPTGOYIIFZNYDEQM.proto' File not found

### DISCUSSION THREAD

**Discussion by geekev** [akvorado/akvorado#1328] [Q&A] [answered]
Hi Vincent,

I am trying to get Akvorado working on Debian without Docker.  The web page loads but I have no flows showing and no 'Last Flow' Record.
I am seeing a similar error to Q&A 1323 except the directory ` /var/lib/clickhouse/format_schemas` is empty.
i.e. we are seeing this error every second in the clickhouse-server.err.log
`2024.07.29 14:09:45.486797 [ 202612 ] {} <Error> StorageKafka (flows_LAABIGYMRYZPTGOYIIFZNYDEQM_raw): void DB::StorageKafka::threadFunc(size_t) Code: 434. DB::Exception: Cannot parse 'flow-LAABIGYMRYZPTGOYIIFZNYDEQM.proto' file, found an error at line -1, column 0, File not found.: While executing Kafka. (CANNOT_PARSE_PROTOBUF_SCHEMA), Stack trace (when copying this message, always include the lines below):
`

I'm working through the troubleshooting and have arrived at the Clickhouse section:
The lag is very big, I tried increasing consumers but I think this probably indicates that nothing has been processed:
<img width="637" alt="akvorado_clickhouse_lag" src="https://github.com/user-attachments/assets/dde67014-8083-4bfb-9d55-1ae58c2f35c9">

These are the tables I have:
<img width="491" alt="akvorado_clickhouse_ltables" src="https://github.com/user-attachments/assets/8eb0efa2-7184-4097-93c9-9230a190c933">

Also, when I restart the orchestrator I get this error `failed to refresh networks dictionary error="code: 1000,` which doesn't seem relevant but I don't really know.

```
./akvorado orchestrator /etc/akvorado/config.yaml
1:41PM WRN akvorado/orchestrator/geoip/root.go:95 > skipping GeoIP component: no database specified module=akvorado/orchestrator/geoip
1:41PM INF akvorado/orchestrator/geoip/root.go:97 > starting GeoIP component module=akvorado/orchestrator/geoip
1:41PM INF akvorado/common/httpserver/root.go:110 > starting HTTP server listen=127.0.0.1:8081 module=akvorado/common/httpserver
1:41PM INF akvorado/common/clickhousedb/root.go:85 > starting ClickHouse component module=akvorado/common/clickhousedb
1:41PM INF akvorado/orchestrator/clickhouse/root.go:97 > starting ClickHouse component module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/common/remotedatasourcefetcher/root.go:127 > starting remote data source fetcher component module=akvorado/common/remotedatasourcefetcher
1:41PM INF akvorado/orchestrator/clickhouse/root.go:166 > ClickHouse component started module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/kafka/root.go:55 > starting Kafka component module=akvorado/orchestrator/kafka
1:41PM INF akvorado/orchestrator/clickhouse/root.go:114 > attempting database migration module=akvorado/orchestrator/clickhouse
1:41PM ??? akvorado/orchestrator/clickhouse/root.go:143 > starting GeoIP refresher module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/kafka/root.go:59 > Kafka component stopped module=akvorado/orchestrator/kafka
1:41PM INF akvorado/cmd/components.go:37 > akvorado has started module=akvorado/cmd version=v1.10.1
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:150 > dictionary asns already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:150 > dictionary protocols already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:150 > dictionary icmp already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:150 > dictionary networks already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:774 > flows_1m0s_consumer already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:774 > flows_5m0s_consumer already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM WRN akvorado/orchestrator/clickhouse/migrations_helpers.go:731 > updating TTL of flows_1h0m0s with interval 1h0m0s, this can take a long time module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:774 > flows_1h0m0s_consumer already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:198 > exporters table already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:248 > exporters view already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:312 > create raw flows table module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:419 > create raw flows consumer view module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:456 > table flows_raw_errors already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations_helpers.go:493 > raw flows errors view already exists, skip migration module=akvorado/orchestrator/clickhouse
1:41PM INF akvorado/orchestrator/clickhouse/migrations.go:171 > database migration done module=akvorado/orchestrator/clickhouse
1:41PM ERR akvorado/orchestrator/clickhouse/networks.go:181 > failed to refresh networks dictionary error="code: 1000, message: Connection refused" module=akvorado/orchestrator/clickhouse
```
Is there a full sample config.yaml file available somewhere and is it possible to find what settings are mandatory , for example do I _have_ to configure BMP for Akvorado to work?

I'd be grateful if you could help me get past these problems.

Regards,
Kevin

**Comment by vincentbernat** [comment, marked as answer]
For the schema, you need to copy it by hand if you are not using Docker. Query `/api/v0/orchestrator/clickhouse/init.sh` for the init script that could do that for you (with the Docker version, we do `curl | sh` for this).

Also, not being able to fetch the dictionary is also a problem, likely to be solved by setting `orchestrator-url` (see https://demo.akvorado.net/docs/configuration#clickhouse).

BMP is not mandatory (and I think you should not use it without a reason). You can get all the available settings with `akvorado orchestrator --dump --check /dev/null`. But the documentation is likely a better reference.

**Reply by geekev** [reply]
Thanks for the ongoing help @vincentbernat ,
 
I've moved past the reported errors , but I am still getting a dictionary error:

`12:40PM ERR akvorado/orchestrator/clickhouse/migrations.go:175 > unable to reload dictionaries after migration error="code: 422, message: Code: 422. DB::Exception: Unsupported scheme in URI 'localhost:8081/api/v0/orchestrator/clickhouse/networks.csv'. (UNSUPPORTED_URI_SCHEME) (version 24.4.1.1733 (official build))" module=akvorado/orchestrator/clickhouse
12:40PM ERR akvorado/orchestrator/clickhouse/networks.go:181 > failed to refresh networks dictionary error="code: 422, message: Unsupported scheme in URI 'localhost:8081/api/v0/orchestrator/clickhouse/networks.csv'" module=akvorado/orchestrator/clickhouse`

I've tried various forms of the clickhouse network config, this is the current test (I've anonymised some of it but the format is the same).
`  networks:
      10.2.3.0/22:
          name: OurCompany
          role: Live
          site: DC
          region: South
          country: UK
          state: London
          city: London
          tenant: Prod
          asn: 99999`

` curl http://localhost:8081/api/v0/orchestrator/clickhouse/networks.csv
network,name,role,site,region,country,state,city,tenant,asn
10.2.3.0/22,OurCompany,Live,Live,DC,South,UK,London,London,Prod,60868`

I'm also seeing these errors continuing beyond the first three:
`12:44PM ERR akvorado/inlet/flow/decoder/netflow/root.go:203 > error while decoding NetFlow/IPFIX error="NetFlowV9 NetFlowV9 [version:9 type:FlowSet obsDomainId:285212672: templateId:0] TemplateSet: reading field [unexpected EOF]"`

Netflow at the moment is coming from a Juniper SRX, we have some MX routers so I may try sending Netflow from one of them as the example in the documentation is an MX.

**Reply by dmbokhan** [reply]
Hi @vincentbernat, how can I get /api/v0/orchestrator/clickhouse/init.sh in v2.0.0? It's no longer in the documentation, as far as I can see.

**Reply by vincentbernat** [reply]
It is not needed anymore as ClickHouse is not using Kafka anymore.

**Comment by broonu** [comment]
Hello Kevin,

If you running the orchestrator in localhost:8080, try this:

`sudo curl http://localhost:8080/api/v0/orchestrator/clickhouse/init.sh | sudo sh`

**Reply by geekev** [reply]
Thanks, yes that did work to get the schema.
